examples/gcp-marketplace/gcp.py

```diff
@@ -0,0 +1,321 @@
+        instance.machine_type = machine_type
+    else:
+        instance.machine_type = f"zones/{zone}/machineTypes/{machine_type}"
+
+    if accelerators:
+        instance.guest_accelerators = accelerators
+
+    if preemptible:
+        # Set the preemptible setting
+        warnings.warn(
+            "Preemptible VMs are being replaced by Spot VMs.", DeprecationWarning
+        )
+        instance.scheduling = compute_v1.Scheduling()
+        instance.scheduling.preemptible = True
+
+    if spot:
+        # Set the Spot VM setting
+        instance.scheduling = compute_v1.Scheduling()
+        instance.scheduling.provisioning_model = (
+            compute_v1.Scheduling.ProvisioningModel.SPOT.name
+        )
+        instance.scheduling.instance_termination_action = instance_termination_action
+
+    if custom_hostname is not None:
+        # Set the custom hostname for the instance
+        instance.hostname = custom_hostname
+
+    if delete_protection:
+        # Set the delete protection bit
+        instance.deletion_protection = True
+
+    # Prepare the request to insert an instance.
+    request = compute_v1.InsertInstanceRequest()
+    request.zone = zone
+    request.project = project_id
+    request.instance_resource = instance
+
+    # Wait for the create operation to complete.
+    print(f"Creating the {instance_name} instance in {zone}...")
+
+    operation = instance_client.insert(request=request)
+
+    wait_for_extended_operation(operation, "instance creation")
+
+    print(f"Instance {instance_name} created.")
+
+    instance = instance_client.get(
+        project=project_id, zone=zone, instance=instance_name
+    )
+
+    return instance
+
+
+def create_from_public_image(
+    project_id: str, zone: str, instance_name: str, machine_type: str, source_image: str
+) -> compute_v1.Instance:
+    """
+    Create a new VM instance with Debian 10 operating system.
+
+    Args:
+        project_id: project ID or project number of the Cloud project you want to use.
+        zone: name of the zone to create the instance in. For example: "us-west3-b"
+        instance_name: name of the new virtual machine (VM) instance.
+
+    Returns:
+        Instance object.
+    """
+
+    disk_type = f"projects/{project_id}/zones/{zone}/diskTypes/pd-balanced"
+    disks = [disk_from_image(disk_type, 10, True, source_image=source_image)]
+    instance = create_instance(project_id, zone, instance_name, disks, machine_type)
+    network_interface = instance.network_interfaces[0]
+    access_configs = network_interface.access_configs
+    print("Your External IP: ", access_configs[0].nat_i_p)
+    return instance
+
+
+if __name__ == "__main__":
+
+    project_id = "<project-id>"
+    zone = "us-central1-c"
+    instance_name = "deepsparse"
+    machine_type = "n2d-highcpu-8"
+    source_image = (
+        "projects/neuralmagic-public/global/images/deepsparse-cloud-142-ubuntu-2204"
+    )
```

Comment:
@InquestGeronimo if these are all that are needed to be changed to update this script for custom usage, could you put these into click arguments? how do you think about zone and machine type fair to default, with project_id as required?

Reply:
@mgoin It is possible to add more variables than the ones found here, the rest are in the create_instance function `def create_instance(
    project_id: str,
    zone: str,
    instance_name: str,
    disks: list[compute_v1.AttachedDisk],
    machine_type: str,
    network_link: str = "global/networks/default",
    subnetwork_link: str = None,
    internal_ip: str = None,
    external_access: bool = True,
    external_ipv4: str = None,
    accelerators: list[compute_v1.AcceleratorConfig] = None,
    preemptible: bool = False,
    spot: bool = False,
    instance_termination_action: str = "STOP",
    custom_hostname: str = None,
    delete_protection: bool = False,`

Reply:
Do you want me to add click args to the 5 variables above and then mention to the users in the README that they can do further customization in the `create_function` if they prefer?